app/controllers/subscribers_auth_token_controller.rb

```diff
@@ -0,0 +1,60 @@
+class SubscribersAuthTokenController < ApplicationController
+  before_action :validate_params
+
+  def auth_token
+    subscriber = find_or_create_subscriber(expected_params.require(:address))
+    token = generate_token(subscriber)
+    email = build_email(subscriber, token)
+
+    DeliveryRequestWorker
+      .perform_async_in_queue(email.id, queue: :delivery_immediate_high)
+
+    render json: { subscriber: subscriber }, status: :created
+  end
+
+private
+
+  def find_or_create_subscriber(address)
```

Comment:
Does this mean if I visit the site to change my subscription and enter an email address that isn't currently in the system, it will create that email address and then send that address a magic link?

Reply:
Yes, that's correct. Not exactly ideal but seems better than getting into the murky world of exposing whether we have subscribers or not by returning different status codes. Does make life a bit painful with activating and deactivating subscribers though :cry:

Reply:
So what happens if someone decides to maliciously enter `n` number of email addresses that don't exist? I'm wondering how that affects our relationship with Notify and whether continually trying to send to non-existent email addresses affects Notify's relationship with Amazon at all?

I'm also a little worried that it really isn't the user's intent to be created in the database at this point in the journey. I wouldn't expect a password reset, for example, to create new details in a database if I use a different email address to the one I signed up with by mistake; and this feels like a similar process. Users wouldn't expect to have the email address stored from this operation. That feels a bit wrong to me. Could we not return a generic "If you're in our database, we've sent you link" to every request?

We also risk confusing and annoying users - if a user signs up to notifications with  `emailaddress1` and then enters `emailaddress2` at this point by mistake, they get an email to their second email address with a link to empty subscriptions.

Also, has anyone looked at this from a GDPR perspective? - I don't know much about it, but seem to remember that it calls for data minimisation.

A minor disadvantage is we potentially end up mis-representing the number of subscribers we have in any stats we create (admittedly outside of bad actors it will be a very minor percentage). But we also have a bunch of orphaned subscribers.

Reply:
Interesting points @bevanloon 👍 

> How does a potential malicious user who decides to enter n number of email addresses that don't exist affect our relationship with Notify, and by extension their relationship with their provider? (i.e. we'll try to send lots of emails to non-existent email addresses).

I'm not too concerned about this as this doesn't introduce this situation. You could do this far worse by signing up to alerts with n number of addresses for instance

> There's a risk here of confusing and annoying users - if a user signs up to notifications with emailaddress1 and then enters emailaddress2 at this point by mistake, they get an email to their second email address with a link to empty subscriptions.

that's true though I'd argue that this is actually less confusing though than a "If you're in our database, we've sent you link" scenario.

> Have we looked at this from a GDPR perspective? - I don't know much about it, but seem to remember that it calls for data minimisation, which I expect one could argue that this isn't.

Not in huge detail. I'd suggest it might be appropriate to that we tell him that their email address is stored.

Where this all gets a bit complicated though and problems emerge with the "If you're in our database, we've sent you link" approach is the whole deactivation process of a user. If you unsubscribe from stuff and then come back later you might find you've not got an account which could be very confusing if you previously had one.

Where I think the sweet spot in this is if we improved the deactivation/nullify process of a user so that every user who interacts with an email does get registered but the process of removing them is more automatic than it currently is.

Reply:
Oh dear. We seem to have backed ourselves into a bit of a corner with all this.

Good point about the signup - that's always been like that though, so we've kept status quo, this is new attack surface for us.

I guess which process is more confusing should come out of user research at some point.

> Where this all gets a bit complicated though and problems emerge with the "If you're in our database, we've sent you link" approach is the whole deactivation process of a user. If you unsubscribe from stuff and then come back later you might find you've not got an account which could be very confusing if you previously had one.

isn't this only the case once you've been nullified though? If you've just been deactivated, we've still got your email address. I agree it is a bit confusing though.